TidepoolKit/Internal/Responses/DataResponse.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  DataResponse.swift
+//  TidepoolKit
+//
+//  Created by Darin Krauss on 2/28/20.
+//  Copyright © 2020 Tidepool Project. All rights reserved.
+//
+
+struct DataResponse: Codable {
+    var data: [TDatum]
+    var malformed: [Int: [String: Any]]
+
+    init(data: [TDatum] = [], malformed: [Int: [String: Any]] = [:]) {
+        self.data = data
+        self.malformed = malformed
+    }
+    
+    init(from decoder: Decoder) throws {
+        self.init()
+        var container = try decoder.unkeyedContainer()
+        while !container.isAtEnd {
+            let superDecoder = try container.superDecoder()
+            let superContainer = try superDecoder.container(keyedBy: CodingKeys.self)
+            do {
+                var datum: TDatum?
+                let type = try superContainer.decode(TDatum.DatumType.self, forKey: .type)
+                switch type {
+                case .cbg:
+                    datum = try TCBGDatum(from: superDecoder)
+                case .food:
+                    datum = try TFoodDatum(from: superDecoder)
+                default:
+                    break // DEPRECATED: Ignore
```

Comment:
Sorry, the comment is misleading. The default case is only because I did not include all of the implemented Tidepool data model types in this PR. Once all types are included (in subsequent PRs) there will be no default case and any unexpected type (aka a new or renamed type) will throw an error when decoding the `enum`. The thrown error will be captured and the "bad" data will be added to the malformed list (unparsable for whatever reason) that can be logged by the client.